lib/bunch.ex

```diff
@@ -236,23 +237,38 @@ defmodule Bunch do
       end)
       |> Enum.group_by(fn {k, _v} -> k end, fn {_k, v} -> v end)
 
-    with_clauses
-    |> Enum.reverse()
-    |> Enum.reduce(block, fn
-      {label, {:<-, meta, _args} = clause}, acc ->
-        label_else_clauses =
-          else_clauses[label] ||
-            "Label `#{inspect(label)}` not present in withl else clauses"
-            |> raise_compile_error(caller, meta)
-
-        {:with, meta, [clause, [do: acc, else: label_else_clauses]]}
-
-      {_label, clause}, acc ->
-        quote do
-          unquote(clause)
-          unquote(acc)
-        end
+    {result, used_labels} =
+      with_clauses
+      |> Enum.reverse()
+      |> Enum.reduce({block, []}, fn
+        {label, {:<-, meta, _args} = clause}, {acc, used_labels} ->
+          label_else_clauses =
+            else_clauses[label] ||
+              "Label `#{inspect(label)}` not present in withl else clauses"
+              |> raise_compile_error(caller, meta)
+
+          {{:with, meta, [clause, [do: acc, else: label_else_clauses]]}, [label | used_labels]}
+
+        {_label, clause}, {acc, used_labels} ->
+          {quote do
+             unquote(clause)
+             unquote(acc)
+           end, used_labels}
+      end)
+
+    unused_else_clauses = Map.drop(else_clauses, used_labels)
+
+    Enum.each(unused_else_clauses, fn {label, clauses} ->
+      Enum.each(clauses, fn {:->, meta, _args} ->
+        log_compile_warning(
```

Comment:
It would be nice to have some tests for it as it would also help me reproduce the warnings